src/GraphQL/TypeExtensions.cs

```diff
@@ -218,11 +221,23 @@ public static Type GetEnumerableElementType(this Type type)
                 }
             }
 
+            // see if the type supports IEnumerable<T>
+            var supportedInterfaces = type.GetInterfaces();
+            foreach (var iface in supportedInterfaces)
+            {
+                if (iface.IsGenericType && iface.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+                {
+                    return iface.GenericTypeArguments[0];
+                }
+            }
+
+            // see if the type supports IEnumerable
+            if (typeof(IEnumerable).IsAssignableFrom(type))
+                return typeof(object);
```

Comment:
If it doesn't make sense to create a `GraphQlClrOutputTypeMapping` for an `object` type, then this code can be removed.  Except it's public, and it would make sense for it to return `object` here -- just not for its use.  So maybe the whole method should be private.  Maybe mark as obsolete now, and make private in v5.  But for now I left it as-is (except expanded the support for `IEnumerable<T>` interfaces).

Reply:
Ok with Obsolete